alpaca/data/historical/option.py

```diff
@@ -0,0 +1,333 @@
+            return raw_latest_quotes
+
+        return parse_obj_as_symbol_dict(Quote, raw_latest_quotes)
+
+    def get_option_latest_trade(
+        self, request_params: OptionLatestTradeRequest
+    ) -> Union[Dict[str, Trade], RawData]:
+        """Retrieves the latest quote for an option symbol or list of option symbols.
+
+        Args:
+            request_params (OptionLatestQuoteRequest): The request object for retrieving the latest quote data.
+
+        Returns:
+            Union[Dict[str, Quote], RawData]: The latest quote in raw or wrapped format
+        """
+        params = request_params.to_request_fields()
+
+        raw_latest_quotes = self._data_get(
+            endpoint_data_type="trades",
+            endpoint_asset_class="options",
+            api_version=self._api_version,
+            extension=DataExtensionType.LATEST,
+            **params,
+        )
+
+        if self._use_raw_data:
+            return raw_latest_quotes
+
+        return parse_obj_as_symbol_dict(Trade, raw_latest_quotes)
+
+    def get_option_snapshot(
+        self, request_params: OptionSnapshotRequest
+    ) -> Union[Dict[str, Snapshot], RawData]:
+        """Returns snapshots of queried symbols. Snapshots contain latest trade and latest quote for the queried symbols.
+
+        Args:
+            request_params (OptionSnapshotRequest): The request object for retrieving snapshot data.
+
+        Returns:
+            Union[SnapshotSet, RawData]: The snapshot data either in raw or wrapped form
+        """
+
+        params = request_params.to_request_fields()
+
+        raw_snapshots = self._data_get(
+            endpoint_asset_class="options",
+            endpoint_data_type="snapshot",
+            api_version=self._api_version,
+            extension=DataExtensionType.SNAPSHOT,
+            **params,
+        )
+
+        if self._use_raw_data:
+            return raw_snapshots
+
+        return parse_obj_as_symbol_dict(Snapshot, raw_snapshots)
+
+    def get_option_chain(
+        self, request_params: OptionChainRequest
+    ) -> Union[Dict[str, Snapshot], RawData]:
+        """The option chain endpoint for underlying symbol provides the latest trade, latest quote for each contract symbol of the underlying symbol.
+
+        Args:
+            request_params (OptionChainRequest): The request object for retrieving snapshot data.
+
+        Returns:
+            Union[SnapshotSet, RawData]: The snapshot data either in raw or wrapped form
+        """
+
+        params = request_params.to_request_fields()
+
+        raw_snapshots = self._data_get(
+            endpoint_asset_class="options",
+            endpoint_data_type="snapshot",
+            api_version=self._api_version,
+            extension=DataExtensionType.SNAPSHOT,
+            **params,
+        )
+
+        if self._use_raw_data:
+            return raw_snapshots
+
+        return parse_obj_as_symbol_dict(Snapshot, raw_snapshots)
+
+    # TODO: Remove duplication
```

Comment:
This should be done!

Reply:
Yeah, please let me refactor this in the followup PR. I have modified a bit for options. Therefore, not exactly matching for stock/crypto/options now.